### ISSUE TITLE
Array Prototype Polyfill causes problems CoffeeScript

### ISSUE THREAD

**Issue by JoeGaebel** [smirzaei/currency-formatter#23]
This [commit](https://github.com/smirzaei/currency-formatter/commit/7ea07c35c0a578b90a352e3ca50ae6e6a29f8b4b) introduces a polyfill which doesn't play nicely with Repos which use CoffeeScript.

In particular, using `of` in CoffeeScript is turns into an `in` in Javascript which loops over keys you don't own.

**Comment by ZeHiro** [comment]
I am not an expert, but it sounds weired. Normally you can include proper js inside coffeescript without any problem.

**Comment by smirzaei** [comment]
I agree with @ZeHiro. I have zero experience with CoffeeScript but shouldn't it play nicely with raw JS?

I think adding the polyfill and removing the dependency on lodash was a good move, but nevertheless we shouldn't break compatibility.

Do you know which part of that polyfill is causing the problem? Maybe we can write it in a different way. 

Hopefully some CoffeeScript experts can help us out, or we can ask a question in Stackoverflow regarding this issue.

**Comment by naganowl** [comment]
In CoffeeScript, you can do something like

```
Array.prototype.foo = ->
a = []
for e of a
  console.log e
```

which turns into

```
var a, e;

Array.prototype.foo = function() {};

a = [];

for (e in a) {
  console.log(e);
}
```

which will log `foo`. This will not happen with `lodash.find`.

**Comment by ZeHiro** [comment]
I see the problem with using `in` or `of`, but here, they are not used in the Polyfill. loadash.find weighs 84k. Isn't the problem a problem of variable scoping?

**Comment by naganowl** [comment]
The problem is when using this library with other code, since we manually added a method to the prototype, when other code executes the native `for..in` loop on an array it will go over `find` where it previously would not have.

**Comment by ZeHiro** [comment]
OK, makes more sense. thanks

**Comment by smirzaei** [comment]
Could you checkout and see if you're still having this issue with the latest version please? Thanks!
